src/metldata/builtin_transformations/common/resolve_path/resolve_path_element_relations.py

```diff
@@ -99,7 +104,12 @@ def resolve_passive_path_element(
     target_resource_ids = set()
 
     for candidate_resource_id, candidate_resource in candidate_resources.items():
-        relation = candidate_resource.relations.get(path_element.property, frozenset())
+        try:
+            relation = candidate_resource.relations[
+                path_element.property
+            ].targetResources
+        except KeyError:
+            relation = frozenset()
```

Comment:
Isn't the logic here basically the same as for `target_resource_ids` above?
Why do this in two different ways?